datafusion/sql/tests/integration_test.rs

```diff
@@ -3390,6 +3390,45 @@ Dml: op=[Update] table=[person]
     prepare_stmt_replace_params_quick_test(plan, param_values, expected_plan);
 }
 
+#[test]
+fn test_prepare_statement_insert_infer() {
+    let sql = "insert into person (id, first_name, last_name) values ($1, $2, $3)";
```

Comment:
Could you add a test for an invalid case as well, such as an out-of-bounds placeholder, so we can see what the UX is like for that?

Reply:
Yep!